### PR TITLE
Quit messages

### PR DESCRIPTION
**Pull request by ObsidianPlague**
The title pretty much says just enough.

### CODE REVIEW

**Comment by Proluiz** [comment]
just to show:
Don't forget:
You're here
forever.

Why would you...

Are you a disappointment?

Press N to keep destroying aliens.
Press Y to surrender.

Please press N. Thank you.

But why?

For every time somebody presses Y,
God kills a puppy.

Y
N

What do you do with your life?

Don't press yes,
there's an army of zombies on your desktop!

All of your friends are dead.
Are you sure you want to quit?


Press Y and a serpentipede
will tear your friggin' face off.

Richard says no.

You come across a minefield.
The sign reads,
"Press Y to jump right in."

Having fun, are we?

**Comment by Calinou** [comment]
I don't understand the `Y\nN` one, what is it supposed to mean? :stuck_out_tongue:

**Comment by Proluiz** [comment]
maybe a generic exit line, i can do two here if you want to:

"They never rest,/n will you rest?"
"If you leave now it's over."

**Comment by chungy** [comment]
It's a good idea, but a not very well-supported one as far as ports go. prboom-plus and chocolate-doom don't support anything beyond the first QUITMSG, for instance and continue to pool in messages from the rest of the hardcoded pool.

**Comment by fragglet** [comment]
I'm in favour of adding replacements for these - it does no harm. However, I think the actual messages need some more work - it doesn't seem like a lot of work was put into them. 

In particular examples like

> Please press N. Thank you.

are actively confusing because it's not clear that the message is a joke unless you've played Doom 1,000 times already and are aware of it.

**Comment by MatthewTheGlutton** [comment]
Ideally every single message should make it clear that pressing Y will terminate the application.

"Do you want to quit this great game?" and "I wouldn't press Y if I were you. Social media is much worse." should be clear enough, but I would submit they should both be considered just barely enough.

**Comment by ObsidianPlague** [comment]
Patch has been updated.

**Comment by MatthewTheGlutton** [comment]
1, 2, 6, 11, 12, 13, 14 still ambiguous.

**Comment by AXDOOMER** [comment]
Here are a few ideas. You're free to make them yours.

- Who will save us if you go away?
- Do you think you can walk away just like that?
- Will you let us all be doomed?
- Press Y to let Hell be brought to Earth.
- If you give up, demons will come and kill us all. 
- Don't quit! Monsters will kill you while you sleep.
- Hey buddy, there's still much work to be done.

**Comment by MatthewTheGlutton** [comment]
Some of those are a bit too Doom (hell, saving people, etc.)
I like these ones:
* Do you think you can walk away just like that?
* Don't quit! Monsters will kill you while you sleep.

I kinda want to like "Hey buddy, there's still much work to be done." but it does leave it a bit ambiguous.

**Comment by AXDOOMER** [comment]
> I kinda want to like "Hey buddy, there's still much work to be done." but it does leave it a bit ambiguous.

I guess that by "ambiguous", you mean making sure that it looks like a quit message. 

How about this?

- Hey buddy, don't go away! There's still much work to be done.

I changed the other ones to be less doomy. 

- Who will stop AGM if you're gone?\nThey still need to be taught a lesson.
- AGM's scientists are planing an even worse nightmare.\nPress Y if you don't care. 
- Press Y to let AGM annihilate Earth with their monstrosities.
- If you give up, this fight will never end.\nWe still need you to restore our freedom.

**Comment by ObsidianPlague** [comment]
> 
> 
> 1, 2, 6, 11, 12, 13, 14 still ambiguous.

14 makes sense because that's a yes & no answer but alright then

**Comment by ObsidianPlague** [comment]
> 
> 
> Some of those are a bit too Doom (hell, saving people, etc.)
> I like these ones:
> 
>     * Do you think you can walk away just like that?
> 
>     * Don't quit! Monsters will kill you while you sleep.
> 
> 
> I kinda want to like "Hey buddy, there's still much work to be done." but it does leave it a bit ambiguous.



> 
> 
> > I kinda want to like "Hey buddy, there's still much work to be done." but it does leave it a bit ambiguous.
> 
> I guess that by "ambiguous", you mean making sure that it looks like a quit message.
> 
> How about this?
> 
>     * Hey buddy, don't go away! There's still much work to be done.
> 
> 
> I changed the other ones to be less doomy.
> 
>     * Who will stop AGM if you're gone?\nThey still need to be taught a lesson.
> 
>     * AGM's scientists are planing an even worse nightmare.\nPress Y if you don't care.
> 
>     * Press Y to let AGM annihilate Earth with their monstrosities.
> 
>     * If you give up, this fight will never end.\nWe still need you to restore our freedom.

Took a couple suggestions, feel free to let me know if something doesn't fit.

File: lumps/dehacked/dehacked.txt [hunk 1]
```diff
@@ -640,20 +640,20 @@ OB_MPTELEFRAG=%o stood where %k was teleporting.
 OB_RAILGUN=%o was railed by %k.
 OB_MPBFG_MBF=%o had %p ass kicked by %k's SKAG!
 # Quit messages.
-QUITMSG = Don't forget:\nYou're here\nforever.
+QUITMSG = AGM's scientists are planning an even worse nightmare.\nPress Y if you don't care.
 QUITMSG1 = Why would you...
-QUITMSG2 = Are you a disappointment?
-QUITMSG3 = Press N to keep destroying aliens.\nPress Y to surrender.
+QUITMSG2 = Hey buddy, there's still much work to be done.\nAre you truly sure you want to quit?
+QUITMSG3 = Press N to keep annihilating aliens.\nPress Y to surrender.
```
Comment (AXDOOMER) [inline lumps/dehacked/dehacked.txt]: `annihilating` is now repeated twice. How about using `decimating` here?

File: lumps/dehacked/dehacked.txt [hunk 2]
```diff
@@ -639,6 +639,22 @@ OB_MPBFG_SPLASH=%o couldn't hide from %k's SKAG.
 OB_MPTELEFRAG=%o stood where %k was teleporting.
 OB_RAILGUN=%o was railed by %k.
 OB_MPBFG_MBF=%o had %p ass kicked by %k's SKAG!
+# Quit messages.
+QUITMSG = AGM's scientists are planning an even worse nightmare.\nPress Y if you don't care.
+QUITMSG1 = Why would you...
+QUITMSG2 = Hey buddy, there's still much work to be done.\nAre you truly sure you want to quit?
+QUITMSG3 = Press N to keep annihilating aliens.\nPress Y to surrender.
```
Comment (AXDOOMER) [inline lumps/dehacked/dehacked.txt]: `annihilating` is now repeated twice. How about using `decimating` here?

Comment (MatthewTheGlutton) [reply lumps/dehacked/dehacked.txt]: Reduce by 1/10?

**Comment by sirocyl** [comment]
Dumb idea: have a condition in the makefile, such that when proper official builds are made, it doesn't do this (the define is set, at the buildserver) but if you're just running `make` on your own machine, it replaces one of the QUITMSGs with something like

> "You're nowhere near finished de-monsterating the code! Git back to work!"

or something similar, poking fun at the developer. ;)

Should be harmless. ¯\\\_(ツ)\_/¯

**Comment by MatthewTheGlutton** [comment]
https://www.urbandictionary.com/define.php?term=decimate

"Obliterate" or "Destroy" would be much better.

**Comment by MatthewTheGlutton** [comment]
No, wait, that's not what I wanted to do.

Here:
`QUITMSG = AGM's developers are planning an even worse nightmare.\nPress Y if you don't care.
QUITMSG1 = Why would anyone press Y to quit such a great game?
QUITMSG2 = Hey buddy, there's still much work to be done.\nAre you truly sure you want to quit?
QUITMSG3 = Press N to keep annihilating aliens.\nPress Y to surrender.
QUITMSG4 = I wouldn't press Y if I were you.\nSocial media is much worse.
QUITMSG5 = If you press Y to leave now, it's over.
QUITMSG6 = We still need you to restore our freedom.\nDo you wish to give up freedom?
QUITMSG7 = They'll never rest.\nWill you rest?
QUITMSG8 = They're all over the place! Let's get out of here!
QUITMSG9 = Don't press Y!\nThere's an army of zombies on your desktop!
QUITMSG10 = Are you sure you want to quit?
QUITMSG11 = Press Y to let AGM overrun Earth with their monstrosities.
QUITMSG12 = Press Y to let the monsters come out under your bed while you sleep.
QUITMSG13 = Do you think you can walk away just like that?
QUITMSG14 = Press Y to do something less worthwhile with your life.
`

**Comment by AXDOOMER** [comment]
Why need to reword everything so that everything clearly implies that either responding "yes" or pressing "Y" will quit the game?

Every source port already displays "Press Y to quit". It's useless and redundant to do it everywhere.

**Comment by MatthewTheGlutton** [comment]
Because some of them were explicit questions where the "Yes" answer implied you wanted to continue playing.

It gave me Bart versus Smokey vibes. https://www.youtube.com/watch?v=wX1x7pfH8fw

**Comment by mc776** [comment]
I just wanna mention SUNPYG's ideas here https://discord.com/channels/616148111076884520/616148111076884524/958956421658075207

They're a bit more meta and less edgelordy, but in a good way that gently emphasizes the "free" part a bit over the "doom" part the way most suggestions on this discussion have gone.

**Comment by mc776** [comment]
I think this one's actually ready to merge...

**Review by mc776 — APPROVED** [review]

**Review by fragglet — APPROVED** [review]

**Pull request merged**